docs/source/tutorial/test_a_project.md

```diff
@@ -0,0 +1,429 @@
+                "price": [120, 290, 30],
+            }
+        )
+
+        duummy_parameters = {
+            "model_options": {
+                "test_size": 0.2,
+                "random_state": 3,
+                "features": ["engines", "passenger_capacity", "crew"],
+            }
+        }
+
+        catalog.add_feed_dict(
+            {
+                "model_input_table" : dummy_data,
+                "params:model_options": dummy_parameters["model_options"],
+            }
+        )
+
+        # Act
+        SequentialRunner().run(pipeline, catalog)
+
+        # Assert
+        assert True     # trivial assert
+
+```
+
+>**NOTE**: `assert True` is a trivial assert statement, included to make the assert step explicit - if the test execution reaches that line the pipeline must have run without error. Tests do not require an assert statement to pass, a test will pass as long as it runs without any errors. This allows this assert statement to be omitted without affecting the validity of the test.
+
+</details>
+
+## Testing best practices
+
+### Where to write your tests
+
+We recommend creating a `tests` directory within the root directory of your project. The structure should mirror the directory structure of `/src/spaceflights`:
+
+```
+src
+│   ...
+└───spaceflights
+│   └───pipelines
+│       └───data_science
+│           │   __init__.py
+│           │   nodes.py
+│           │   pipeline.py
+│
+tests
+|   ...
+└───pipelines
+    |    test_data_science_pipeline.py
+```
+
+
+### Using fixtures
+
+In our tests, we can see that `dummy_data` and `dummy_parameters` have been defined three times with (mostly) the same values. Instead, we can define these outside of our tests as [pytest fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#fixture):
+
+<details>
+
+```python
+import pytest
+
+@pytest.fixture
+def dummy_data():
+    return pd.DataFrame(
+        {
+            "engines": [1, 2, 3],
+            "crew": [4, 5, 6],
+            "passenger_capacity": [5, 6, 7],
+            "price": [120, 290, 30],
+        }
+    )
+
+@pytest.fixture
+def dummy_parameters():
+    parameters = {
+        "model_options": {
+            "test_size": 0.2,
+            "random_state": 3,
+            "features": ["engines", "passenger_capacity", "crew"],
+        }
+    }
+    return parameters
+```
+
+</details>
+
+We can then access these through the test arguments.
+
+```python
+def test_split_data(dummy_data, dummy_parameters):
+        ...
+```
+
+### Pipeline slicing
+
+In the test `test_data_science_pipeline` we test the that outputs of the data science pipeline are those that are expected. However, as pipelines are not static, this test is not robust. Instead we should be specific with how we define the pipeline to be tested; we do this by using [pipeline slicing](../nodes_and_pipelines/slice_a_pipeline.md#slice-a-pipeline-by-running-specified-nodes) to specify the pipeline's start and end:
```

Comment:
"we test the that outputs of the data science pipeline are those that are expected", at the moment we don't do this, but test that the pipeline runs without errors, that's why you needed to add the trivial `assert` statement. I think it would be better if we could indeed check some sort of output, even if it's just a logging statement.